### PR TITLE
port tests for F.perspective and transforms.RandomPerspective

### PR DESCRIPTION
**Pull request by pmeier**
cc @vfdev-5

### CODE REVIEW

File: torchvision/transforms/v2/_geometry.py [hunk 1]
```diff
@@ -1008,8 +1008,8 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
         return self._call_kernel(
             F.perspective,
             inpt,
-            None,
-            None,
+            startpoints=None,
+            endpoints=None,
```
Comment (pmeier) [inline torchvision/transforms/v2/_geometry.py:1012]: Hard to read without keyword arguments.

File: torchvision/transforms/v2/functional/_geometry.py [hunk 2]
```diff
@@ -1548,7 +1548,7 @@ def _perspective_image_pil(
     image: PIL.Image.Image,
     startpoints: Optional[List[List[int]]],
     endpoints: Optional[List[List[int]]],
-    interpolation: Union[InterpolationMode, int] = InterpolationMode.BICUBIC,
+    interpolation: Union[InterpolationMode, int] = InterpolationMode.BILINEAR,
```
Comment (pmeier) [inline torchvision/transforms/v2/functional/_geometry.py:1551]: We have used bicubic here, but bilinear for the tensor kernels, since that is what v1 does:

https://github.com/pytorch/vision/blob/e13b8f5c3616bdc58fa847a848d63acdd416a692/torchvision/transforms/_functional_pil.py#L321

https://github.com/pytorch/vision/blob/e13b8f5c3616bdc58fa847a848d63acdd416a692/torchvision/transforms/_functional_tensor.py#L704

However, since the PIL kernel is private in v1, this divergence doesn't really matter, since they are always called through the functional

https://github.com/pytorch/vision/blob/e13b8f5c3616bdc58fa847a848d63acdd416a692/torchvision/transforms/functional.py#L719

https://github.com/pytorch/vision/blob/e13b8f5c3616bdc58fa847a848d63acdd416a692/torchvision/transforms/functional.py#L758-L762

Although in v2 the PIL kernel is private as well I see no reason for this divergence. LMK if you think otherwise.

File: test/test_transforms_v2_refactored.py [hunk 3]
```diff
@@ -3421,6 +3422,9 @@ def test_kernel_mask_errors(self, fill):
         with pytest.raises(ValueError, match="Non-scalar fill value is not supported"):
             check_kernel(F.pad_mask, make_segmentation_mask(), padding=[1], fill=fill)
 
+    def test_kernel_video(self):
```
Comment (pmeier) [inline test/test_transforms_v2_refactored.py:3425]: Driveby. I forgot this in #7939.

File: torchvision/transforms/v2/functional/_geometry.py (hunk 2 above)
Comment (NicolasHug) [reply torchvision/transforms/v2/functional/_geometry.py:1551]: Good catch, thanks

File: test/test_transforms_v2_refactored.py [hunk 4]
```diff
@@ -3514,3 +3518,245 @@ def test_bounding_boxes_correctness(self, padding, format, dtype, device, fn):
...
+                endpoints=None,
+                coefficients=coefficients,
+                interpolation=interpolation,
+                fill=fill,
+            )
+        )
+
+        if interpolation is transforms.InterpolationMode.BILINEAR:
+            mae = (actual.float() - expected.float()).abs().mean()
+            assert mae < 3
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:3674]: This will help me sleep better and makes sure most values are within 1 pixel difference. I only tested in on CPU locally, hopefully it also works on GPUs

```suggestion
            abs_diff = (actual.float() - expected.float()).abs()
            assert (abs_diff > 1).float().mean() < .07

            mae = abs_diff.mean()
            assert mae < 3
```

File: torchvision/transforms/v2/functional/_geometry.py (hunk 2 above)
Comment (NicolasHug) [reply torchvision/transforms/v2/functional/_geometry.py:1551]: Sooooo this needs a cherry-pick?

Comment (pmeier) [reply torchvision/transforms/v2/functional/_geometry.py:1551]: Not really. PIL kernel is private for both v1 and v2. And thus their default values do not matter, but only the default values of the functional that calls them, since that overwrites the default. And that is bilinear and thus aligned with the public tensor kernels.

Comment (NicolasHug) [reply torchvision/transforms/v2/functional/_geometry.py:1551]: oh right it's private, all good then. Thanks!

File: test/test_transforms_v2_refactored.py (hunk 4 above)
Comment (pmeier) [reply test/test_transforms_v2_refactored.py]: We are testing against PIL here. Meaning, there is no GPU comparison.
